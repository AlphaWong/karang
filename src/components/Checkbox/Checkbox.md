#### Usage:

```js static
import { Checkbox } from 'lalamove-ui';
```

```js
<Checkbox label="Checkbox" disabled />
```
