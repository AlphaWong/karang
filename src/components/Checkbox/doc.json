{
  "name": "Checkbox",
  "import": "import { Checkbox } from 'lalamove-ui'"
}
