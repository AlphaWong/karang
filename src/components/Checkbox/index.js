import React from 'react';
import { string, bool, func } from 'prop-types';
import styled from 'styled-components';

import noop from 'utils/noop';
import { orange, gray, silver, white, offWhite } from 'styles/colors';
import { primaryFonts, fontSize } from 'styles/fonts';

const Text = styled.span`
  padding-left: 28px;
`;

const Checkmark = styled.span`
  border: solid 1px ${silver};
  height: 16px;
  width: 16px;
  position: absolute;

  &:after {
    content: '';
    display: none;
    position: absolute;
    border: solid ${orange};
    border-width: 0 2px 2px 0;
    height: 9px;
    width: 4px;
    left: 5px;
    top: 1px;
    transform: rotate(45deg);
  }
`;

const Input = styled.input`
  width: 1px;
  height: 1px;
  padding: 0;
  border: 0;
  margin: -1px;
  clip: rect(0, 0, 0, 0);
  overflow: hidden;
  position: absolute;

  &:focus ~ ${Checkmark} {
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.22);
  }

  &:checked ~ ${Checkmark} {
    background-color: ${white};
  }

  &:checked ~ ${Checkmark}:after {
    display: block;
  }

  &:disabled ~ ${Checkmark} {
    background-color: ${offWhite};
  }

  &:disabled ~ ${Checkmark}:after {
    border-color: ${silver};
  }

  &:disabled ~ ${Text} {
    opacity: 0.7;
  }
`;

const Container = styled.label`
  position: relative;
  font-family: ${primaryFonts};
  color: ${gray};
  cursor: pointer;
  display: inline-flex;
  align-items: center;
  font-size: ${fontSize.regular};
  user-select: none;

  &:hover ${Input} ~ ${Checkmark} {
    border: solid 1px ${silver};
  }

  &:hover ${Checkmark} {
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.22);
  }
`;

const CheckBox = ({
  checked,
  name,
  label,
  onChange,
  disabled,
  ...remainProps
}) => (
  <Container htmlFor={name} {...remainProps}>
    <Input
      type="checkbox"
      name={name}
      id={name}
      checked={checked}
      onChange={onChange}
      disabled={disabled}
    />
    <Checkmark />
    <Text>{label}</Text>
  </Container>
);

CheckBox.propTypes = {
  name: string.isRequired, // eslint-disable-line
  label: string,
  checked: bool,
  onChange: func,
  disabled: bool,
};

CheckBox.defaultProps = {
  label: '',
  checked: false,
  onChange: noop,
  disabled: false,
};

export default CheckBox;
