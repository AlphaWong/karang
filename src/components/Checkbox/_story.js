import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs/react';
import Checkbox from './index';
//
// class Wrapper extends Component {
//   state = {
//     0: { toggle: true }, // eslint-disable-line react/no-unused-state
//     1: { toggle: true, disabled: true }, // eslint-disable-line react/no-unused-state
//   };
//
//   handleToggle = i => {
//     this.setState(state => ({
//       ...state,
//       [i]: { ...!state[i], toggle: !state[i].toggle },
//     }));
//   };
//
//   render() {
//     return Object.entries(this.state).map(([key, cb]) => (
//       <Fragment>
//         <Checkbox
//           checked={cb.toggle}
//           onChange={() => this.handleToggle(key)}
//           name={`demo${key}_checkbox`}
//           label={`demo${key}`}
//           disabled={cb.disabled}
//         />
//         <br />
//         <br />
//       </Fragment>
//     ));
//   }
// }

storiesOf('Checkbox', module).add('Basic', () => (
  <Checkbox label="Checkbox" disabled={boolean('disabled', false)} />
));
