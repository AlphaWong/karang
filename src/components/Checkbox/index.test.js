/* global render */
import React from 'react';
import Checkbox from './index';

test('not crash', () => {
  expect(() => <Checkbox />).not.toThrow();
});

test('snapshot', () => {
  expect(render(<Checkbox />)).toMatchSnapshot();
});
