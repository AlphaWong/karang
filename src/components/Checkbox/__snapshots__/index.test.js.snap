// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c1 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(88,89,91,1);
  font-family: 'Noto Sans',sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 20px;
}

.c3 {
  position: absolute;
  overflow: hidden;
  width: 1px;
  height: 1px;
  padding: 0;
  border: 0;
  margin: -1px;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
}

.c5 {
  padding-left: 28px;
  cursor: pointer;
}

.c2:disabled ~ .c5 {
  opacity: 0.7;
  cursor: not-allowed;
}

.c4 {
  position: absolute;
  width: 16px;
  height: 16px;
  border: solid 1px rgba(128,130,133,1);
  cursor: pointer;
}

.c4:after {
  content: '';
  position: absolute;
  top: 1px;
  left: 5px;
  display: none;
  width: 4px;
  height: 9px;
  border: solid rgba(242,103,34,1);
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c2:focus ~ .c4 {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
}

.c2:checked ~ .c4 {
  background-color: rgba(255,255,255,1);
}

.c2:checked ~ .c4:after {
  display: block;
}

.c2:disabled ~ .c4 {
  background-color: rgba(229,229,229,1);
  cursor: not-allowed;
}

.c2:disabled ~ .c4:after {
  border-color: rgba(128,130,133,1);
}

.c0:hover .c2:not(:disabled) ~ .c4 {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
  border: solid 1px rgba(128,130,133,1);
}

<label
  class="c0 c1"
  for="llm-checkbox"
>
  <input
    class="c2 c3"
    id="llm-checkbox"
    name="llm-checkbox"
    type="checkbox"
  >
    
  </input>
  <span
    class="c4"
  >
    
  </span>
  <span
    class="c5"
  >
    
  </span>
</label>
`;
